koku/api/user_settings/views.py

```diff
@@ -56,38 +59,43 @@ def __post_init__(self):
         else:
             raise SettingsInvalidFilterException("Invalid not a user setting")
 
-    def update_cost_type(self, request_data, schema_name):
-        serializer = UserSettingUpdateCostTypeSerializer(schema_name, request_data)
+    def update_cost_type(self, schema_name):
+        serializer = UserSettingUpdateCostTypeSerializer(schema_name, self.request.data)
         if serializer.is_valid(raise_exception=True):
-            set_cost_type(schema_name, request_data.get(self.setting))
+            set_cost_type(schema_name, self.request.data.get(self.setting))
             invalidate_view_cache_for_tenant_and_source_type(schema_name, Provider.PROVIDER_AWS)
             return Response(status=status.HTTP_204_NO_CONTENT)
 
-    def update_currency(self, request_data, schema_name):
-        serializer = UserSettingUpdateCurrencySerializer(schema_name, request_data)
+    def update_currency(self, schema_name):
+        serializer = UserSettingUpdateCurrencySerializer(schema_name, self.request.data)
         if serializer.is_valid(raise_exception=True):
-            set_currency(schema_name, request_data.get(self.setting))
+            set_currency(schema_name, self.request.data.get(self.setting))
             invalidate_view_cache_for_tenant_and_all_source_types(schema_name)
             return Response(status=status.HTTP_204_NO_CONTENT)
 
-    def retrieve_user_settings(self, request):
-        users_setting = {"settings": {self.setting: self.get_param(request)}}
+    def retrieve_user_settings(self):
```

Comment:
Maybe use `get_` instead of `retrieve_`.
```suggestion
    def get_user_settings(self):
```